### ISSUE TITLE
False Negative of `clippy::manual_strip`

### ISSUE THREAD

**Issue by yhx-12243** [rust-lang/rust-clippy#12863]
### Summary

Clippy does not report a `manual_strip` warning.

### Lint Name

manual_strip

### Reproducer

I tried this code:

```rust
pub fn f(mut x: &str) -> usize {
    if x.starts_with(' ') {
        x = &x[1..];
    }
    x.len()
}
```

I expected to see this happen:

```
warning: stripping a prefix manually
 --> 1.rs:3:9
  |
3 |         &x[1..]
  |         ^^^^^^^
  |
note: the prefix was tested here
 --> 1.rs:2:5
  |
2 |     if x.starts_with(' ') {
  |     ^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_strip
  = note: `#[warn(clippy::manual_strip)]` on by default
help: try using the `strip_prefix` method
  |
2 ~     if let Some(<stripped>) = x.strip_prefix(' ') {
3 ~         x = <stripped>;
  |
```

Instead, this happened:

<nothing>

### Version

```text
rustc 1.80.0-nightly (84b40fc90 2024-05-27)
binary: rustc
commit-hash: 84b40fc908c3adc7e0e470b3fbaa264df0e122b8
commit-date: 2024-05-27
host: aarch64-apple-darwin
release: 1.80.0-nightly
LLVM version: 18.1.6
```

**Comment by AurelienFT** [comment]
The lint isn't trigger because it seems that there is [a filter to don't lint if the `&str` is mutable](https://github.com/rust-lang/rust-clippy/blob/e7efe4381af5ae065105f6fa8c30ae86e01d3d9a/clippy_lints/src/manual_strip.rs#L92). I still didn't found the reason behind this. I still trying to understand, if someone have an idea in the meantime, feel free to comment

**Comment by yhx-12243** [comment]
> The lint isn't trigger because it seems that there is [a filter to don't lint if the `&str` is mutable](https://github.com/rust-lang/rust-clippy/blob/e7efe4381af5ae065105f6fa8c30ae86e01d3d9a/clippy_lints/src/manual_strip.rs#L92). I still didn't found the reason behind this. I still trying to understand, if someone have an idea in the meantime, feel free to comment

Reasonable since `x` may be modified before slicing, making it hard to detect whether it is a manual strip.

**Comment by AurelienFT** [comment]
Make sense. I think it still could be done but I'm too new on this repository for this.

@rustbot unassign
